connectors/src/connectors/microsoft/temporal/activities.ts

```diff
@@ -616,20 +627,24 @@ function removeAllButLastOccurences(deltaList: microsoftgraph.DriveItem[]) {
  * The function makes the assumption that there is no circular parent
  * relationship
  */
-function sortForIncrementalUpdate(changedList: DriveItem[], rootId: string) {
+function sortForIncrementalUpdate(changedList: DriveItem[], rootId?: string) {
   if (changedList.length === 0) {
     return [];
   }
 
   const internalIds = changedList.map((item) => getDriveItemInternalId(item));
 
   const sortedItemList = changedList.filter((item) => {
-    if (item.id === rootId) {
+    if (!item.parentReference) {
```

Comment:
would move that test below the 2 ones rootId / !rootId
if for some reason we don't have the parent reference but it is the correct rootId or full drive, we still want to include

Reply:
i was needing it  because of item.parentReference.id , but shold be ok with item.root